src/analyses/libraryFunctions.ml

```diff
@@ -167,23 +167,34 @@ let invalidate_actions = [
     "GetSpinlock", readsAll;
     "ReleaseSpinlock", readsAll;
     "atoi", readsAll;             (*safe*)
+    "__builtin_ctz", readsAll;
+    "__builtin_ctzl", readsAll;
+    "__builtin_ctzll", readsAll;
+    "__builtin_clz", readsAll;
     "bzero", writes [1]; (*keep 1*)
     "connect", readsAll;          (*safe*)
     "fclose", readsAll;           (*safe*)
     "fflush", writesAll;          (*unsafe*)
     "fopen", readsAll;            (*safe*)
+    "fdopen", writes [1];            (*safe*)
+    "setvbuf", writes[2];
```

Comment:
`int setvbuf(FILE *stream, char *buf, int mode, size_t size);`
According to [this](https://www.cplusplus.com/reference/cstdio/setvbuf/), this sets `buf` to be the output buffer for operation on the `FILE *stream*`. That would mean that any future write to (or `flush` of) `stream` could cause a write to the buffer.
Basically we would have to treat the `buffer` as top after such a call. Maybe add a TODO?

Reply:
I am not really aware what the type behind `FILE` is, and how it is represented in Goblint, but it may be that `stream` is actually also written?

Reply:
I changed it such that is written to, but my suspicion is that we don't really do anything terribly useful with them anyway.